src/main/java/com/google/firebase/messaging/LightSettings.java

```diff
@@ -96,16 +99,20 @@ public Builder setLightOnDurationInMillis(long lightOnDurationInMillis) {
      * @return This builder.
      */
     public Builder setLightOffDurationInMillis(long lightOffDurationInMillis) {
-      checkArgument(lightOffDurationInMillis >= 0, "ttl must not be negative");
-      long seconds = TimeUnit.MILLISECONDS.toSeconds(lightOffDurationInMillis);
+      this.lightOffDuration = convertToSecondsAndNanosFormat(lightOffDurationInMillis);
+      return this;
+    }
+
+    private String convertToSecondsAndNanosFormat(long millis) {
+      checkArgument(millis >= 0, "ttl must not be negative");
```

Comment:
Milliseconds duration must not be negative

Reply:
Done